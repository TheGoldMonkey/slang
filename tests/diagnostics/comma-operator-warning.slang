//DIAGNOSTIC_TEST:SIMPLE:

int testCommaOperator()
{
    // This should generate a warning - comma operator in variable initialization
    float4 vColor = (0.f, 0.f, 0.f, 1.f);
    
    // This should NOT generate a warning - comma operator in for-loop side effect
    for (int i = 0; i < 10; i++, vColor.x++)
    {
    }
    
    // This should generate a warning - comma operator in regular expression
    int x = (1, 2, 3);
    
    // This should NOT generate a warning - comma operator in return statement
    int a = 5;
    return a *= 2, a + 1;
}

void someFunction(int value) {}