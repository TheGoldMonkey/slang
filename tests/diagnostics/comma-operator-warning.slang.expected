result code = 0
standard error = {
tests/diagnostics/comma-operator-warning.slang(6): warning 41024: comma operator used in expression (may be unintended)
    float4 vColor = (0.f, 0.f, 0.f, 1.f);
                        ^
tests/diagnostics/comma-operator-warning.slang(6): warning 41024: comma operator used in expression (may be unintended)
    float4 vColor = (0.f, 0.f, 0.f, 1.f);
                             ^
tests/diagnostics/comma-operator-warning.slang(6): warning 41024: comma operator used in expression (may be unintended)
    float4 vColor = (0.f, 0.f, 0.f, 1.f);
                                  ^
tests/diagnostics/comma-operator-warning.slang(14): warning 41024: comma operator used in expression (may be unintended)
    int x = (1, 2, 3);
              ^
tests/diagnostics/comma-operator-warning.slang(14): warning 41024: comma operator used in expression (may be unintended)
    int x = (1, 2, 3);
                 ^
}
standard output = {
}